modules/artifact/src/main/java/org/mule/runtime/module/artifact/api/descriptor/AbstractArtifactDescriptorFactory.java

```diff
@@ -218,12 +218,19 @@ private BundleDescriptor getBundleDescriptor(File appFolder, M artifactModel) {
     }
 
     try {
-      return bundleDescriptorLoader.load(appFolder, artifactModel.getBundleDescriptorLoader().getAttributes(), getArtifactType());
+      return bundleDescriptorLoader
+          .load(appFolder, getBundleDescriptorAttributes(artifactModel.getBundleDescriptorLoader(), deploymentProperties),
+                getArtifactType());
     } catch (InvalidDescriptorLoaderException e) {
       throw new ArtifactDescriptorCreateException(e);
     }
   }
 
+  protected Map<String, Object> getBundleDescriptorAttributes(MuleArtifactLoaderDescriptor bundleDescriptorLoader,
+                                                              Optional<Properties> deploymentPropertiesOptional) {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Remove this unused method parameter "deploymentPropertiesOptional". ([squid:S1172](https://sonarqube.kbuild.msap.io/coding_rules#rule_key=squid%3AS1172))

[See it in SonarQube](https://sonarqube.kbuild.msap.io/project/issues?id=mule-runtime-mule&pullRequest=7687&issues=AWj0M4ipotIXnmsUdjY6&open=AWj0M4ipotIXnmsUdjY6)

[ISSUE_KEY]: AWj0M4ipotIXnmsUdjY6